api/server/router/local/container.go

```diff
@@ -419,14 +420,25 @@ func (s *router) postContainersAttach(ctx context.Context, w http.ResponseWriter
 		fmt.Fprintf(outStream, "HTTP/1.1 200 OK\r\nContent-Type: application/vnd.docker.raw-stream\r\n\r\n")
 	}
 
+	keys := []byte{}
+	detachKeys := r.FormValue("detachKeys")
+	if detachKeys != "" {
+		keys, err = term.ToBytes(detachKeys)
+		if err != nil {
+			logrus.Warnf("Wrong escape keys provided (%s) using default : ctrl-p ctrl-q", detachKeys)
```

Comment:
Check my understanding, this checks the string that is coming in can be parsed as key codes.

If that is the true case, I would suggest that the text would be more accurate as "Invalid escape keys provided". My point is, it is not the keys that are wrong, it is the incoming string.